### ISSUE TITLE
DS data accessibility fixes

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2911]
### Story

As a user working on the DS data page
I want the inputs to have clear instructions 
so that I can avoid mistakes that could bring down my website

### Acceptance Criteria

- [ ] The changes match these [designs](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9908-19094&t=TmnelYl74IuMUrEv-4).
- [ ] New view state and Edit DS data primary button
- [ ] Delete button is moved next to the h2.
- [ ] Primary and secondary buttons order is switched
- [ ] Content updates for view and edit state
- [ ] New hint text for Key tag and Digest inputs
- [ ] New modal when Cancel button is pressed
- [ ] New review page

**Comment by gabydisarli** [comment]
Turned this into a sub-issue of #2368

**Comment by abroddrick** [comment]
@gabydisarli were the designs for this updated after the nameservers ticket was finished?

**Comment by gabydisarli** [comment]
@abroddrick no, its still blocked

**Comment by katypies** [comment]
This is an accessibility issue and a high impact, but it's not a result of our latest compliance audit, and therefore is not labeled for compliance. It's also quite large, and so not suited for the accessibility pod work.

**Comment by SamiyahKey** [comment]
This has been unblocked by #3067

**Comment by gabydisarli** [comment]
@SamiyahKey  the ds data page redesign actually doesn’t reflect the new patterns we’ve created in the name servers page. It has issues with the technical parts that we encountered on the name servers page that need to be fixed before it can be implemented. I'm going to put blocked back on.

**Comment by katypies** [comment]
This ticket needs to be rewritten/updated to encompass the redesign work needed (design side) + spinning up a dev ticket.

**Comment by gabydisarli** [comment]
@katypies @Katherine-Osos @witha-c I've created a new sub-issue to capture the design work still needed here.

**Comment by allly-b** [comment]
@gabo0oo name servers isn't done yet, so we were blocking until name servers is done and merged.

**Comment by allly-b** [comment]
I'm removing refinement since this ticket was refined and now is just blocked by the nameservers ticket
